### PR TITLE
Fixes and cleanups for Keil MDK examples

### PR DESCRIPTION
**Pull request by dgarske**
# Description

Fixes and cleanups for Keil MDK examples.
Use SP math all by default.
Disable old/deprecated algorithms.
Add CMSIS RTOS timer support for TLS and benchmark.

ZD 15981

# Testing

Keil MDK5.38, wolfSSL Keil CMSIS pack and Infineon NuMaker M2354 (Cortex M23 at 96MHz).

# Checklist

 - [ ] added tests
 - [ ] updated/added doxygen
 - [ ] updated appropriate READMEs
 - [ ] Updated manual and documentation

### CODE REVIEW

**Comment by lealem47** [comment]
Changes look good. It would be nice to consolidate all these `user_settings.h` files. They look largely identical except for `MDK_CONF_NETWORK` set to 0 for the Crypt examples.

**Comment by lealem47** [comment]
Jenkins retest this please

**Comment by dgarske** [comment]
@lealem47 good suggestion about combining the user_settings.h. Can you run the CMSIS pack creator and test this out with Keil MDK?

**Comment by dgarske** [comment]
Retest this please

**Comment by dgarske** [comment]
@lealem47 provided me the updated Keil CMSIS pack based on this PR and I was able to confirm the user_settings.h refactor works. Tested loading the sample projects and creating new projects with pack.

**Comment by dgarske** [comment]
@kojo1 , can you also review this PR?

**Comment by dgarske** [comment]
Retest this please

File: IDE/MDK5-ARM/Conf/user_settings.h [hunk 1]
```diff
@@ -21,21 +21,20 @@
 
 #define NO_MAIN_DRIVER
 #define BENCH_EMBEDDED
-#define NO_DEV_RANDOM
-#define WOLFSSL_USER_CURRTIME
 #define SIZEOF_LONG_LONG 8
 #define NO_WRITEV
 #define NO_DEV_RANDOM
+#define WOLFSSL_IGNORE_FILE_WARN
```
Comment (lealem47) [inline IDE/MDK5-ARM/Conf/user_settings.h:27]: I had to define `NO_INLINE` to get it building

File: IDE/MDK5-ARM/README.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+## To install the wolfSSL pack
+
+1) Open the Keil "Pack Installer"
+2) Under Generic locate "wolfSSL::wolfSSL"
+3) Click "Install"
+
+## To add the pack to your project
+
+1) Project -> Manage -> "Runtime Environment"
+2) Expand "wolfSSL" and check the boxes for wolfCrypt CORE and wolfSSL CORE.
```
Comment (lealem47) [inline IDE/MDK5-ARM/README.md:16]: Mention that by default the pack assumes the CMSIS RTOS v2 and Keil TCP network packs are enabled

**Review by lealem47 — CHANGES_REQUESTED** [review]
user_settings refactor looks great!

File: IDE/MDK5-ARM/Conf/user_settings.h (hunk 1 above)
Comment (dgarske) [reply IDE/MDK5-ARM/Conf/user_settings.h:27]: I did not. That's very odd. Can you please share the errors you were seeing?

Comment (lealem47) [reply IDE/MDK5-ARM/Conf/user_settings.h:27]:
```
C:/Users/Owner/AppData/Local/Arm/Packs/wolfSSL/wolfSSL/5.6.2/wolfssl/wolfcrypt/test/test.c(36): warning: In file included from...
C:/Users/Owner/AppData/Local/Arm/Packs/wolfSSL/wolfSSL/5.6.2/wolfssl\wolfssl/wolfcrypt/mem_track.h(149): error: unknown type name 'inline'
static WC_INLINE void* TrackMalloc(size_t sz)
       ^
C:/Users/Owner/AppData/Local/Arm/Packs/wolfSSL/wolfSSL/5.6.2/wolfssl\wolfssl/wolfcrypt/types.h(320): note: expanded from macro 'WC_INLINE'
                   #define WC_INLINE inline
                                     ^
C:/Users/Owner/AppData/Local/Arm/Packs/wolfSSL/wolfSSL/5.6.2/wolfssl/wolfcrypt/test/test.c(36): warning: In file included from...
C:/Users/Owner/AppData/Local/Arm/Packs/wolfSSL/wolfSSL/5.6.2/wolfssl\wolfssl/wolfcrypt/mem_track.h(226): error: unknown type name 'inline'
static WC_INLINE void TrackFree(void* ptr)
       ^
C:/Users/Owner/AppData/Local/Arm/Packs/wolfSSL/wolfSSL/5.6.2/wolfssl\wolfssl/wolfcrypt/types.h(320): note: expanded from macro 'WC_INLINE'
                   #define WC_INLINE inline
                                     ^
C:/Users/Owner/AppData/Local/Arm/Packs/wolfSSL/wolfSSL/5.6.2/wolfssl/wolfcrypt/test/test.c(36): warning: In file included from...
C:/Users/Owner/AppData/Local/Arm/Packs/wolfSSL/wolfSSL/5.6.2/wolfssl\wolfssl/wolfcrypt/mem_track.h(237): warning: cast from 'byte *' (aka 'unsigned char *') to 'memoryTrack *' (aka 'struct memoryTrack *') increases required alignment from 1 to 4 [-Wcast-align]
    mt = (memoryTrack*)((byte*)ptr - sizeof(memoryTrack));
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
C:/Users/Owner/AppData/Local/Arm/Packs/wolfSSL/wolfSSL/5.6.2/wolfssl\wolfssl/wolfcrypt/mem_track.h(300): error: unknown type name 'inline'
static WC_INLINE void* TrackRealloc(void* ptr, size_t sz)
       ^
C:/Users/Owner/AppData/Local/Arm/Packs/wolfSSL/wolfSSL/5.6.2/wolfssl\wolfssl/wolfcrypt/types.h(320): note: expanded from macro 'WC_INLINE'
                   #define WC_INLINE inline
                                     ^
C:/Users/Owner/AppData/Local/Arm/Packs/wolfSSL/wolfSSL/5.6.2/wolfssl\wolfssl/wolfcrypt/mem_track.h(791): error: unknown type name 'inline'
static WC_INLINE void StackTrap(void)
       ^
C:/Users/Owner/AppData/Local/Arm/Packs/wolfSSL/wolfSSL/5.6.2/wolfssl\wolfssl/wolfcrypt/types.h(320): note: expanded from macro 'WC_INLINE'
                   #define WC_INLINE inline
                                     ^
```

Comment (lealem47) [reply IDE/MDK5-ARM/Conf/user_settings.h:27]: I didn't change the default compiler for the Keil µVision project
```
-xc -std=c90 --target=arm-arm-none-eabi -mcpu=cortex-m23 
```

Comment (lealem47) [reply IDE/MDK5-ARM/Conf/user_settings.h:27]: Nope. I'll look if it's getting turned on somewhere for the platform

Comment (lealem47) [reply IDE/MDK5-ARM/Conf/user_settings.h:27]: But regardless if inline is an unknown type, might as well disable it
```
 error: unknown type name 'inline'
```

Comment (lealem47) [reply IDE/MDK5-ARM/Conf/user_settings.h:27]: Looks like it's complaining about the WC_INLINE in the function declaration in `mem_track.h` and that declaration in the header is not guarded with `WOLFSSL_TRACK_MEMORY`

**Review by lealem47 — APPROVED** [review]

**Review by lealem47 — COMMENTED** [review]
Awesome

**Review by lealem47 — APPROVED** [review]

**Pull request merged**